sign/sign_integration_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sign_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/release-sdk/sign"
+	"sigs.k8s.io/release-sdk/test"
+)
+
+func generateCosignKey(t *testing.T, getPassFunc cosign.PassFunc) string {
+	tempDir, err := os.MkdirTemp("", "k8s-cosign-keys-")
+	require.Nil(t, err)
+
+	keys, err := cosign.GenerateKeyPair(getPassFunc)
+	require.Nil(t, err)
+	require.NotNil(t, keys)
+
+	err = os.WriteFile(filepath.Join(tempDir, "cosign.key"), keys.PrivateBytes, 0600)
+	require.Nil(t, err)
+
+	err = os.WriteFile(filepath.Join(tempDir, "cosign.pub"), keys.PublicBytes, 0644)
+	require.Nil(t, err)
+
+	return tempDir
+}
```

Comment:
Do we want to extend the interface to support generating keys? Do we want to create some test interface that would extend the default interface?

Reply:
Which interface, the public API? I would say no in this case because we do not need it in production.

Reply:
Yes, the one defined here: https://github.com/kubernetes-sigs/release-sdk/blob/main/sign/impl.go

In that case, I guess it makes sense to leave it as a helper function here.